components/clp-package-utils/clp_package_utils/controller.py

```
@@ -0,0 +1,773 @@
+
+def get_or_create_instance_id(clp_config: CLPConfig) -> str:
+    """
+    Gets or creates a unique instance ID for this CLP instance.
+
+    :param clp_config:
+    :return: The instance ID.
+    """
+    instance_id_file_path = clp_config.logs_directory / "instance-id"
+
+    if instance_id_file_path.exists():
+        with open(instance_id_file_path, "r") as f:
```

Ruff (0.14.0)

Unnecessary mode argument

Remove mode argument

(UP015)